viadot/sources/business_core.py

```diff
@@ -0,0 +1,142 @@
+import pandas as pd
+import json
+from prefect.utilities import logging
+from typing import Any, Dict
+
+from ..exceptions import CredentialError, APIError
+from .base import Source
+from ..config import local_config
+from ..utils import handle_api_response
+
+
+logger = logging.get_logger(__name__)
+
+
+class BusinessCore(Source):
+    """
+    Source for getting data from Bussines Core ERP API.
+
+    """
+
+    def __init__(
+        self,
+        url: str = None,
+        filters_dict: Dict[str, Any] = {
+            "BucketCount": None,
+            "BucketNo": None,
+            "FromDate": None,
+            "ToDate": None,
+        },
+        verify: bool = True,
+        credentials: Dict[str, Any] = None,
+        config_key: str = "BusinessCore",
+        *args,
+        **kwargs,
+    ):
+        """
+        Creating an instance of BusinessCore source class.
+
+        Args:
+            url (str, optional): Base url to a view in Business Core API. Defaults to None.
+            filters_dict (Dict[str, Any], optional): Filters in form of dictionary. Available filters: 'BucketCount',
+                'BucketNo', 'FromDate', 'ToDate'.  Defaults to None.
+            verify (bool, optional): Whether or not verify certificates while connecting to an API. Defaults to True.
+            credentials (Dict[str, Any], optional): Credentials stored in a dictionary. Required credentials: username,
+                password. Defaults to None.
+            config_key (str, optional): Credential key to dictionary where details are stored. Defaults to "BusinessCore".
+
+        Raises:
+            CredentialError: When credentials are not found.
+        """
+        DEFAULT_CREDENTIALS = local_config.get(config_key)
+        credentials = credentials or DEFAULT_CREDENTIALS
+
+        required_credentials = ["username", "password"]
+        if any([cred_key not in credentials for cred_key in required_credentials]):
+            not_found = [c for c in required_credentials if c not in credentials]
+            raise CredentialError(f"Missing credential(s): '{not_found}'.")
+
+        self.credentials = credentials
```

Comment:
`self.credentials` are created after `super().__init__` so this line is not needed here.